src/devices/bus/pce_ctrl/pachinko.h

```diff
@@ -0,0 +1,62 @@
+// license:BSD-3-Clause
+// copyright-holders:kmg
+/**********************************************************************
+
+    NEC PC Engine Coconuts Japan CJPC-101 Pachinko Controller
+
+**********************************************************************/
+
+#ifndef MAME_BUS_PCE_CTRL_PACHINKO_H
+#define MAME_BUS_PCE_CTRL_PACHINKO_H
+
+#pragma once
+
+
+#include "pcectrl.h"
+
+
+
+//**************************************************************************
+//  TYPE DEFINITIONS
+//**************************************************************************
+
+// ======================> pce_pachinko_device
+
+class pce_pachinko_device : public device_t,
+							public device_pce_control_port_interface
+{
+public:
+	// construction/destruction
+	pce_pachinko_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock);
+
+	// optional information overrides
+	virtual ioport_constructor device_input_ports() const override;
+
+protected:
+	// device-level overrides
+	virtual void device_start() override;
+	virtual void device_reset() override;
+
+	// device_pce_control_port_interface overrides
+	virtual u8 peripheral_r() override;
+	virtual void sel_w(int state) override;
+	virtual void clr_w(int state) override;
+
+private:
+	// internal states
+	u8 m_counter = 0;
+	bool m_prev_sel = false; // previous SEL pin state
+	bool m_prev_clr = false; // previous CLR pin state
```

Comment:
Please don’t initialise device members in headers – do it in the constructor initialiser list.  Doing it in the header has multiple drawbacks:
* More to preprocess in downstream things that `#include` the header.
* More to recompile in the event initialiser values are changed.
* When initialisation is split between the constructor and the declarations in the header, you need to look in two places.
* It gets confusing if a conflicting initialiser is added to the constructor without removing the initialiser from the header.

Reply:
Or just put it in device_start, excluding things that may get changed at mcfg time.